pkg/providers/instancetype/nodetemplate_test.go

```diff
@@ -516,6 +516,65 @@ var _ = Describe("NodeTemplate/InstanceTypes", func() {
 		}
 		Expect(nodeNames.Len()).To(Equal(1))
 	})
+	It("should launch instances for AWS NeuronCore resource requests", func() {
+		nodeNames := sets.NewString()
+		ExpectApplied(ctx, env.Client, provisioner, nodeTemplate)
+		pods := []*v1.Pod{
+			coretest.UnschedulablePod(coretest.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("4")},
+					Limits:   v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("4")},
+				},
+			}),
+			// Should pack onto same instance
+			coretest.UnschedulablePod(coretest.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("32")},
+					Limits:   v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("32")},
+				},
+			}),
+			// Should pack onto a separate instance
+			coretest.UnschedulablePod(coretest.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("64")},
+					Limits:   v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("64")},
+				},
+			}),
+		}
+		ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pods...)
+		for _, pod := range pods {
+			node := ExpectScheduled(ctx, env.Client, pod)
+			Expect(node.Labels).To(HaveKeyWithValue(v1.LabelInstanceTypeStable, "inf1.24xlarge"))
+			nodeNames.Insert(node.Name)
+		}
+		Expect(nodeNames.Len()).To(Equal(2))
+	})
+	It("should launch trn1 instances for AWS Neuron resource requests", func() {
+		nodeNames := sets.NewString()
+		provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{
+			{
+				Key:      v1.LabelInstanceTypeStable,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"trn1.2xlarge"},
+			},
+		}
+		ExpectApplied(ctx, env.Client, provisioner, nodeTemplate)
+		pods := []*v1.Pod{
+			coretest.UnschedulablePod(coretest.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1alpha1.ResourceAWSNeuronCore: resource.MustParse("2")},
```

Comment:
Did you intend to test requesting the "neurondevice" resource request in one of these tests?

Reply:
Yes, I realize that's a miss. I'll add them to the nodeclass test.